api/v1alpha1/ratelimitfilter_types.go

```diff
@@ -99,6 +99,11 @@ type RateLimitSelectCondition struct {
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
 	Headers []HeaderMatch `json:"headers,omitempty"`
+
+	// SourceIP is IP CIDR from `X-Forwarded-For` header.
```

Comment:
want to ensure that edge proxy case is taken care of, will the internal implementation add the downstream's source IP into `X-Forwarded-For` by setting `use_remote_address` ?https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#x-forwarded-for

Reply:
I do not think, there's something special.

Reply:
imho this would need to be applied internally else the user might not be able to ratelimit based on the  L3 Source IP

Reply:
you may need implement a new action in Envoy?

Reply:
why not peruse existing mechanism highlighted above ? setting `use_remote_address` which ensures XFF has remote/client/source IP set

Reply:
will raise a GH issue once this PR is in, to ensure `use_remote_address` is set

Reply:
> will raise a GH issue once this PR is in, to ensure `use_remote_address` is set

xref: https://github.com/envoyproxy/gateway/issues/998

Reply:
thanks for raising it, we should also reframe the current doc string into - 
```
// SourceIP is the IP CIDR that represents the range of Source IP Addresses of the client. These could also be the intermediate addresses through which the request has flown through and is part of the  `X-Forwarded-For` header.
```